File: go/pkg/repl/repl.go
package repl

import (
	"bufio"
	"fmt"
	"io"

	"github.com/aidanaden/monkey-lang/pkg/lexer"
	"github.com/aidanaden/monkey-lang/pkg/token"
)

const PROMPT = ">> "

func Start(in io.Reader, out io.Writer) {
	scanner := bufio.NewScanner(in)

	for {
		fmt.Print(PROMPT)
		if scanned := scanner.Scan(); !scanned {
			return
		}

		line := scanner.Text()
		lex := lexer.New(line)

		for tok := lex.NextToken(); tok.Type != token.EOF; tok = lex.NextToken() {
			fmt.Printf("%+v\n", tok)
		}
	}
}

File: rust/src/repl/repl.rs
use std::io::{self, Write};

use crate::lexer::{lexer::Lexer, token::Token};

pub fn start() {
    loop {
        print!(">> ");
        let _ = io::stdout().flush();
        let mut lines = io::stdin().lines();
        if let Some(Ok(line)) = lines.next() {
            let mut lex = Lexer::new(line);
            while let Ok(tok) = lex.next_token() {
                println!("{}", tok);
                if tok == Token::Eof {
                    break;
                }
            }
        }
    }
}
